client/network/src/protocol/sync.rs

```diff
@@ -507,7 +507,7 @@ impl<B: BlockT> ChainSync<B> {
 
 	/// Number of active sync requests.
 	pub fn num_sync_requests(&self) -> usize {
-		self.fork_targets.len()
+		self.fork_targets.values().filter(|f| f.number <= self.best_queued_number).count()
```

Comment:
Why do we need to filter here?

Reply:
This function is used in tests to detect when there's no activity (`block_unti_idle`). Since now `fork_targets` might contain some future blocks, that are not requested until we reach that block number it will stall tests. Basically only forks that can be requested are considered as active.

Reply:
Ahh okay, ty. Maybe we should add some comment to the function to explain this :D